Engine/Component/ComponentCanvas.cpp

```diff
@@ -1,38 +1,127 @@
 #include "ComponentCanvas.h"
 
+#include "EditorUI/Panel/PanelGame.h"
+#include "Component/ComponentCanvasRenderer.h"
 #include "Main/Application.h"
 #include "Main/GameObject.h"
 
+#include "Module/ModuleCamera.h"
+#include "Module/ModuleEditor.h"
+#include "Module/ModuleScene.h"
 #include "Module/ModuleUI.h"
+#include "Module/ModuleWindow.h"
 
-ComponentCanvas::ComponentCanvas() : ComponentUI(ComponentUI::UIType::CANVAS)
+#include <queue>
+
+
+ComponentCanvas::ComponentCanvas() : Component(ComponentType::CANVAS)
 {
 	
 }
 
-ComponentCanvas::ComponentCanvas(GameObject* owner) : ComponentUI(owner, ComponentUI::UIType::CANVAS)
+ComponentCanvas::ComponentCanvas(GameObject* owner) : Component(owner, ComponentType::CANVAS)
 {
 	
 }
 
-void ComponentCanvas::Render(float4x4* projection) 
+ComponentCanvas& ComponentCanvas::operator=(const ComponentCanvas& component_to_copy)
+{
+	Component::operator = (component_to_copy);
+	return *this;
+}
+
+Component* ComponentCanvas::Clone(bool original_prefab) const
 {
-	owner->transform_2d.SetSize(App->ui->window_width, App->ui->window_height);
-	ComponentUI::Render(projection);
+	ComponentCanvas* created_component;
+	if (original_prefab)
+	{
+		created_component = new ComponentCanvas();
+	}
+	else
+	{
+		created_component = App->ui->CreateComponentCanvas();
+	}
+	*created_component = *this;
+	return created_component;
+};
+
+void ComponentCanvas::Copy(Component* component_to_copy) const
+{
+	*component_to_copy = *this;
+	*static_cast<ComponentCanvas*>(component_to_copy) = *this;
 }
 
 void ComponentCanvas::Delete()
 {
-	ComponentUI::Delete();
+	App->ui->RemoveComponentCanvas(this);
 }
 
-void ComponentCanvas::Save(Config& config) const
+void ComponentCanvas::Render(bool scene_mode)
```

Comment:
True means that the camera has to take in account that is rendering for the editor `PanelScene`. I use this to render the canvas on the world using a perspective frustum instead of the projection one.